client/package.json

```diff
@@ -3,106 +3,97 @@
-    "eslint-plugin-jsx-a11y": "^6.4.1",
-    "eslint-plugin-prettier": "^4.0.0",
-    "eslint-plugin-react": "^7.24.0",
-    "eslint-plugin-react-hooks": "^4.2.0",
-    "eslint-plugin-styled-components-a11y": "^0.0.34",
-    "improved-yarn-audit": "^2.3.3",
-    "jest-image-snapshot": "^4.5.0",
-    "jest-localstorage-mock": "^2.4.14",
-    "jest-puppeteer": "^5.0.4",
-    "jest-styled-components": "^7.0.4",
-    "jest-when": "^3.3.1",
-    "nock": "^13.1.0",
-    "prettier": "2.3.1",
-    "puppeteer": "^10.0.0",
-    "redux-mock-store": "^1.5.4",
-    "serve": "^12.0.0",
-    "styled-components": "~5.3.5",
-    "stylelint": "^13.13.1",
-    "stylelint-a11y": "^1.2.3",
-    "stylelint-config-prettier": "^8.0.2",
-    "stylelint-config-recommended": "^5.0.0",
-    "stylelint-config-styled-components": "^0.1.1",
-    "stylelint-processor-styled-components": "^1.10.0",
-    "typescript": "^4.4.3",
-    "typescript-plugin-styled-components": "^2.0.0"
+    "@eslint/js": "9.15.0",
+    "@eslint/migrate-config": "^1.3.4",
+    "@stylistic/eslint-plugin": "2.11.0",
+    "@testing-library/dom": "10.4.0",
+    "@testing-library/jest-dom": "6.6.3",
+    "@testing-library/react": "16.0.1",
+    "@testing-library/react-hooks": "8.0.1",
+    "@testing-library/user-event": "14.5.2",
+    "@types/bootstrap": "5.2.10",
+    "@types/jest-image-snapshot": "6.4.0",
+    "@types/leaflet": "1.9.14",
+    "@types/node": "22.9.1",
+    "@types/react": "18.3.12",
+    "@types/react-dom": "18.3.1",
+    "@types/react-paginate": "7.1.4",
+    "@types/react-redux": "7.1.34",
+    "@types/react-router-hash-link": "2.4.9",
+    "@types/sanitize-html": "2.13.0",
+    "@types/styled-components": "5.1.34",
+    "@types/wellknown": "0.5.8",
+    "@typescript-eslint/eslint-plugin": "8.15.0",
+    "@typescript-eslint/parser": "8.15.0",
+    "eslint": "9.15.0",
+    "eslint-config-prettier": "9.1.0",
+    "eslint-plugin-import": "2.31.0",
+    "eslint-plugin-jsx-a11y": "6.10.2",
+    "eslint-plugin-prettier": "5.2.1",
+    "eslint-plugin-react": "7.37.2",
+    "eslint-plugin-react-hooks": "5.0.0",
+    "eslint-plugin-styled-components-a11y": "2.1.36",
+    "jest-image-snapshot": "6.4.0",
+    "nock": "13.5.6",
+    "prettier": "3.3.3",
+    "redux-mock-store": "1.5.5",
+    "serve": "14.2.4",
+    "styled-components": "6.1.13",
+    "stylelint": "16.10.0",
+    "stylelint-a11y": "1.2.3",
+    "stylelint-config-prettier": "9.0.5",
+    "stylelint-config-recommended": "14.0.1",
+    "stylelint-config-standard": "36.0.1",
+    "ts-jest": "29.2.5",
+    "typescript": "5.7.2",
+    "typescript-plugin-styled-components": "3.0.0",
+    "vitest": "2.1.5"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject",
+    "start": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
     "lint": "yarn lint:eslint && yarn lint:stylelint",
```

Comment:
We'd better remove "yarn lint:stylelint" and just run "yarn lint:eslint" for now. It is currently broken, and without the "stylelint-processor-styled-components" processor, which seems to be hopelessly out-of-date now, there is no way to check our styled-components code anyway.

Reply:
Changed/removed